### PR TITLE
small fixes to hltDiff

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

A few small fixes to `hltDiff`, to better handle incorrect command-line arguments:

* added a check on the content of `old_config` (otherwise, an invalid process name, via `-N` or `-O`, can lead to a seg-fault);

* improved checks on validity of output `.root`/`.csv`/`.json` files (if one specifies as output `-F foo/bar`, and the output folder `foo` does not exist already, the outputs will not be created; in such cases, the executable was giving a seg-fault (`csv` outputs), or erroneously stating that the output was created (`json` files)).

The final diff is rather confusing due to code-format; it might be easier to review these few commits individually.

#### PR validation:

Validated with private tests.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

File: HLTrigger/Tools/bin/hltDiff.cc [hunk 1]
```diff
@@ -747,49 +747,63 @@ class JsonOutputProducer {
...
         out_file << indent(1) << key("events") << '[';  // line open
-        for (auto it = v_events.begin(); it != v_events.end(); ++it) {
-          out_file << (*it).serialise(2);
-          if (it != --v_events.end())
-            out_file << ',';
-        }
+        // for (std::vector<JsonEvent>::const_iterator it = v_events.begin(); it != v_events.end(); ++it) {
+        //   out_file << (*it).serialise(2);
+        //   if (it != --v_events.end()) out_file << ',';
+        // }
```
Comment (fwyzard) [inline HLTrigger/Tools/bin/hltDiff.cc:782]: @missirol should these commented-out lines be uncommented, kept, or removed ?

Comment (missirol) [reply HLTrigger/Tools/bin/hltDiff.cc:782]: Hi @fwyzard , not sure. I didn't touch those (despite what the diff suggests..); I think they are just [these lines](https://github.com/cms-sw/cmssw/pull/33174/files#diff-7a488f2c7fdc3b0f6ccdaab9428d9a61632af34963fc1b8dfbf17bd77d61d220L775); indentation changes make it look like they were changed, but they weren't. From the looks of it, they could be probably removed.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
Append ?w=1 at the end of https://github.com/cms-sw/cmssw/pull/33174/files to get a diff ignoring white-space changes:
https://github.com/cms-sw/cmssw/pull/33174/files?w=1

**Comment by Martin-Grunewald** [comment]
+1

**Comment by missirol** [comment]
> Append ?w=1 at the end of https://github.com/cms-sw/cmssw/pull/33174/files to get a diff ignoring white-space changes:
> https://github.com/cms-sw/cmssw/pull/33174/files?w=1

I see. Thanks, Martin!

**Comment by qliphy** [comment]
+1

**Pull request merged**